#
# Ubuntu: none
# Upstream: none
# Description: fix FTBFS with cairo 1.8.4 that is built with directfb,
#  triggering some new code, even when directfb was not wanted/detected
#  in configure
#
---
 gfx/thebes/src/gfxASurface.cpp |    4 ++++
 1 file changed, 4 insertions(+)

Index: mozilla/gfx/thebes/src/gfxASurface.cpp
===================================================================
--- mozilla.orig/gfx/thebes/src/gfxASurface.cpp
+++ mozilla/gfx/thebes/src/gfxASurface.cpp
@@ -50,19 +50,21 @@
 #include "gfxXlibSurface.h"
 #endif
 
 #ifdef CAIRO_HAS_QUARTZ_SURFACE
 #include "gfxQuartzSurface.h"
 #include "gfxQuartzImageSurface.h"
 #endif
 
+#ifdef MOZ_DFB
 #ifdef CAIRO_HAS_DIRECTFB_SURFACE
 #include "gfxDirectFBSurface.h"
 #endif
+#endif
 
 #ifdef CAIRO_HAS_QPAINTER_SURFACE
 #include "gfxQPainterSurface.h"
 #endif
 
 #include <stdio.h>
 #include <limits.h>
 
@@ -166,21 +168,23 @@
 #ifdef CAIRO_HAS_QUARTZ_SURFACE
     else if (stype == CAIRO_SURFACE_TYPE_QUARTZ) {
         result = new gfxQuartzSurface(csurf);
     }
     else if (stype == CAIRO_SURFACE_TYPE_QUARTZ_IMAGE) {
         result = new gfxQuartzImageSurface(csurf);
     }
 #endif
+#ifdef MOZ_DFB
 #ifdef CAIRO_HAS_DIRECTFB_SURFACE
     else if (stype == CAIRO_SURFACE_TYPE_DIRECTFB) {
         result = new gfxDirectFBSurface(csurf);
     }
 #endif
+#endif
 #ifdef CAIRO_HAS_QPAINTER_SURFACE
     else if (stype == CAIRO_SURFACE_TYPE_QPAINTER) {
         result = new gfxQPainterSurface(csurf);
     }
 #endif
     else {
         result = new gfxUnknownSurface(csurf);
     }
